io/disk/arcconf/arcconf_cntl_oper.py

```diff
@@ -0,0 +1,136 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>
+
+
+"""
+arcconf - Array configuration utility for PMC-Sierra
+(Microsemi) Controllers.
+"""
+
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado import main
+from avocado.utils import distro
+
+
+class Arcconftest(Test):
+
+    """
+    Covers functionality relavent to controller settings.
+    """
+
+    def setUp(self):
+
+        """
+        Checking if the required packages are installed,
+        if not found specific packages will be installed.
+        """
+        smm = SoftwareManager()
+        if not smm.check_installed("lsscsi") and not smm.install("lsscsi"):
+            self.skip("Unable to install lsscsi")
+        self.crtl_no = self.params.get('crtl_no')
+        self.pci_id = self.params.get('pci_id', default="").split(",")
+        self.http_path = self.params.get('http_path')
+        self.tool_name = self.params.get('tool_name')
+        self.option = self.params.get("option")
+        self.option_args = self.params.get("option_args")
+
+        # Gets the list of PCIIDs on the system
+        cmd = 'for device in $(lspci | awk \'{print $1}\') ; do echo \
+              $(lspci -vmm -nn -s $device | grep "\[" | awk \'{print $NF}\' \
+              | sed -e "s/\]//g" | sed -e "s/\[//g" | tr \'\\n\' \' \' \
+              | awk \'{print $2,$3,$4,$5}\') ; done'
+        pci_id = self.cmdop_list(cmd)
+        pci_id_formatted = []
+        for i in pci_id.splitlines():
+            pci_id_formatted.append(str(i.replace(" ", ":")))
+
+        # check if all the yaml parameters are entered
+        if self.crtl_no is '' or self.pci_id is '' or self.tool_name \
+           is '' or self.http_path is '':
+            self.skip(" Test skipped!!, please ensure yaml parameters are \
+            entered or disk count is more than 1")
+        elif self.comp(self.pci_id, pci_id_formatted) == 1:
+            self.skip(" Test skipped!!, PMC controller not available")
+
+        detected_distro = distro.detect()
+        if not smm.check_installed("arcconf"):
+            if detected_distro.name == "Ubuntu":
+                http_repo = "%s%s.deb" % (self.http_path, self.tool_name)
+                self.repo = self.fetch_asset(http_repo, expire='10d')
+                cmd = "dpkg -i %s" % self.repo
+            else:
+                http_repo = "%s%s.rpm" % (self.http_path, self.tool_name)
+                self.repo = self.fetch_asset(http_repo, expire='10d')
+                cmd = "rpm -ivh %s" % self.repo
+            if process.run(cmd, shell=True) == 0:
```

Comment:
i think this condition not going to match as process.run always return a object please use` process.system()`